pennylane/hf/molecule.py

```diff
@@ -0,0 +1,153 @@
+from pennylane.hf.basis_data import atomic_numbers
+from pennylane.hf.basis_set import BasisFunction, mol_basis_data
+
+
+class Molecule:
+    r"""Create a molecule object that stores molecular information and default basis set parameters.
+
+    The molecule object can be passed to functions that perform a Hartree-Fock calculation.
+
+    Args:
+        symbols (list[str]): symbols of the atomic species in the molecule
+        coordinates (array[float]): 1D array with the atomic positions in Cartesian coordinates. The
+            coordinates must be given in atomic units and the size of the array should be ``3*N``
+             where ``N`` is the number of atoms.
+        charge (int): Net charge of the molecule. If not specified, a neutral molecule is assumed.
+        mult (int): Spin multiplicity :math:`\mathrm{mult}=N_\mathrm{unpaired} + 1` for
+            :math:`N_\mathrm{unpaired}` unpaired electrons occupying the HF orbitals. Possible
+            values of ``mult`` are :math:`1, 2, 3, \ldots`. If not specified, a closed-shell HF
+            state is assumed.
+        basis_name(str): Atomic basis set used to represent the molecular orbitals.
+        active_electrons (int): Number of active electrons. If not specified, all electrons are
+            considered to be active.
+        active_orbitals (int): Number of active orbitals. If not specified, all orbitals are
+            considered to be active.
+        l (tuple[int]): angular momentum numbers of the basis function.
+        alpha (array(float)): exponents of the primitive Gaussian functions
+        coeff (array(float)): coefficients of the contracted Gaussian functions
+        r (array(float)): positions of the Gaussian functions
+    """
+
+    def __init__(
+        self,
+        symbols,
+        coordinates,
+        charge=0,
+        mult=1,
+        basis_name="sto-3g",
+        active_electrons=None,
+        active_orbitals=None,
+        l=None,
+        alpha=None,
+        coeff=None,
+        r=None,
+    ):
+
+        if basis_name not in ["sto-3g", "STO-3G"]:
+            raise ValueError("The only supported basis set is 'sto-3g'.")
+
+        if set(symbols) - set(atomic_numbers):
+            raise ValueError(f"Atoms in {set(symbols) - set(atomic_numbers)} are not supported.")
+
+        self.symbols = symbols
+        self.coordinates = coordinates
+        self.charge = charge
+        self.mult = mult
+        self.basis_name = basis_name
+
+        self.n_basis, self.basis_data = mol_basis_data(self.basis_name, self.symbols)
+
+        if l is None:
+            l = [i[0] for i in self.basis_data]
+
+        if alpha is None:
+            alpha = [pnp.array(i[1], requires_grad=False) for i in self.basis_data]
+
+        if coeff is None:
+            coeff = [pnp.array(i[2], requires_grad=False) for i in self.basis_data]
+
+        if r is None:
+            r = sum([[self.coordinates[i]] * self.n_basis[i] for i in range(len(self.n_basis))], [])
```

Comment:
How come there is an empty list at the end here?

Reply:
It was there to create a list of lists, instead of a list of lists of lists. I have used a more understandable format using itertools.

Reply:
The new form is clearer I agree 👍